DataFormats/EgammaReco/src/classes_def.xml

```diff
@@ -86,11 +88,19 @@
   <class name="edm::RefVector<std::vector<reco::PreshowerClusterShape>,reco::PreshowerClusterShape,edm::refhelper::FindUsingAdvance<std::vector<reco::PreshowerClusterShape>,reco::PreshowerClusterShape> >"/>
   <class name="edm::Wrapper<edm::RefVector<std::vector<reco::PreshowerClusterShape>,reco::PreshowerClusterShape,edm::refhelper::FindUsingAdvance<std::vector<reco::PreshowerClusterShape>,reco::PreshowerClusterShape> > >" splitLevel="0"/>
 
-  <class name="reco::ElectronSeed" ClassVersion="12">
+  <class name="reco::ElectronSeed" ClassVersion="13">
+   <version ClassVersion="13" checksum="3112955595"/> 
    <version ClassVersion="12" checksum="536071409"/>
    <version ClassVersion="11" checksum="3831394066"/>
    <version ClassVersion="10" checksum="534580602"/>
   </class>
+  <ioread sourceClass="reco::ElectronSeed" version="[1-12]" targetClass="reco::ElectronSeed" source="float dRz2_; float dPhi2_; float dRz2Pos_; float dPhi2Pos_; float dRz1_; float dPhi1_; float dRz1Pos_; float dPhi1Pos_; unsigned char hitsMask_;" target="hitInfo_">
+    <![CDATA[hitInfo_ = reco::ElectronSeed::createHitInfo(onfile.dPhi1Pos_,onfile.dPhi1_,onfile.dRz1Pos_,onfile.dRz1_,onfile.dPhi2Pos_,onfile.dPhi2_,onfile.dRz2Pos_,onfile.dRz2_,onfile.hitsMask_,newObj->recHits());]]>
```

Comment:
Hi Andrea, this was definitely something I struggled with. 

However my understanding is that the new object is already created and this code is just setting hitInfo_ to the correct value. 

it happens line 2986-2995:
`vector<reco::ElectronSeed::PMVars>& hitInfo_ = *(vector<reco::ElectronSeed::PMVars>*)(target+offset_hitInfo_);
reco::ElectronSeed* newObj = (reco::ElectronSeed*)target;
hitInfo_ = reco::ElectronSeed::createHitInfo(onfile.dPhi1Pos_,onfile.dPhi1_,onfile.dRz1Pos_,onfile.dRz1_,onfile.dPhi2Pos_,onfile.dPhi2_,onfile.dRz2Pos_,onfile.dRz2_,onfile.hitsMask_,newObj->recHits());'

I feel if target was pointing to an invalid location , hitInfo_ would also be bad. And newObj cant be null, this function is definitely being called and is using the newObj. Wouldnt it segfault in that case? If that data was corrupted, my hit mask and sub det ids would be messed up when reading back. I dont think they are:
https://sharper.web.cern.ch/sharper/cms/egamma/eleSeedValidation/readingOldSeedsV2/

Still you raise a worrying point and I'll keep digging to be 100% sure. I can probably do oldObj but its not straightforward at all (its not a reco::ElectronSeed*)